share/models/ingest.py

```diff
@@ -122,18 +121,11 @@ def get_transformer(self):
     def acquire_lock(self, using='locking'):
         # NOTE: Must be in transaction
         logger.debug('Attempting to lock %r', self)
-        try:
-            with connections[using].cursor() as cursor:
-                cursor.execute('''
-                    SELECT "id"
-                    FROM "{}"
-                    WHERE id = %s
-                    FOR NO KEY UPDATE NOWAIT;
-                '''.format(self._meta.db_table), (self.id,))
-        except DatabaseError:
-            logger.warning('Lock failed; another task is already harvesting %r.', self)
-            raise HarvesterConcurrencyError('Unable to lock {!r}'.format(self))
-        else:
+        with connections[using].cursor() as cursor:
+            cursor.execute('''SELECT pg_try_advisory_lock('{}'::regclass::integer, {});'''.format(self._meta.db_table, self.id))
```

Comment:
`self.id` should be passed as a parameter to `cursor.execute` to fill in a `%s`. Actually, it looks like you could do that with `db_table` as well.

Any reason that you're double casting (`::regclass::integer`) `self.id`?